src/Raven.Server/ServerWide/CompareExchangeExpirationStorage.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            if (value.TryGetMember(Constants.Documents.Metadata.Key, out var metadata))
+            {
+                if (metadata is BlittableJsonReaderObject bjro && bjro.TryGet(Constants.Documents.Metadata.Expires, out object obj))
+                {
+                    if (obj is LazyStringValue expirationDate)
+                    {
+                        if (DateTime.TryParseExact(expirationDate, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind, out DateTime date) == false)
+                        {
+                            var inner = new InvalidOperationException(
+                                $"The expiration date format for compare exchange '{CompareExchangeKey.SplitStorageKey(storageKey).Key}' is not valid: '{expirationDate}'. Use the following format: {DateTime.UtcNow:O}");
+                            throw new RachisApplyException("Could not apply command.", inner);
+                        }
+                        var expiry = date.ToUniversalTime();
+                        ticks = expiry.Ticks;
+                        return true;
+                    }
+                    else
+                    {
+                        var inner = new InvalidOperationException($"The type of {Constants.Documents.Metadata.Expires} metadata for compare exchange '{CompareExchangeKey.SplitStorageKey(storageKey).Key}' is not valid. Use the following type: {nameof(DateTime)}");
+                        throw new RachisApplyException("Could not apply command.", inner);
+                    }
+                }
+            }
+
+            ticks = default;
+            return false;
+        }
+
+        public static unsafe bool DeleteExpiredCompareExchange(ClusterOperationContext context, Table items, long ticks, long take = long.MaxValue)
+        {
+            using (ExpiredCleaner(context))
+            {
+                foreach (var tuple in GetExpiredValues(context, ticks))
+                {
+                    if (take-- <= 0)
+                        return true;
+
+                    if (items.ReadByKey(tuple.keySlice, out var reader) == false)
+                        continue;
+
+                    var storeValue = reader.Read((int)ClusterStateMachine.CompareExchangeTable.Value, out var size);
+                    using var result = new BlittableJsonReaderObject(storeValue, size, context);
+
+                    if (HasExpiredMetadata(tuple.keySlice.ToString(), result, out long currentTicks) == false)
+                        continue;
+                    if (currentTicks > tuple.expiredTicks)
+                        continue;
+
+                    items.Delete(reader.Id);
+                }
+
+                return false;
+            }
+        }
+
+        public static IDisposable ExpiredCleaner(ClusterOperationContext context)
+        {
+            _expired = new Dictionary<Slice, List<Slice>>();
```

Comment:
This should not be a static field, or a field at all.
It should be something that you pass in the context.